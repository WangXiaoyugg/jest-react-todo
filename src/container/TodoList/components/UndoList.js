import React, {Component} from 'react';
import "../style.css";

class UndoList extends Component {
    constructor(props) {
        super(props);

    }


    render() {
        console.log("this.props", this.props);
        const list = this.props.list;
        console.log("list:", list);

        return (
            <div>
                <div data-test="count">{list.length}</div>
                <ul>
                    {
                        list.map((item, index) => {
                            return (
                                <li data-test="list-item" key={index}>
                                    {item}
                                    <span data-test="delete-item">-</span>
                                </li>
                            )
                        })
                    }
                </ul>
            </div>
        );
    }

}

export default UndoList;
