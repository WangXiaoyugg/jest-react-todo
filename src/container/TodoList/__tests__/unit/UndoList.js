import React from 'react';
import  {shallow} from 'enzyme';

import UndoList from '../../components/UndoList';
import { findTestWrapper } from '../../../../utils/testUtils'

it('UndoList 渲染样式正常', () => {
    const wrapper = shallow(<UndoList list={[]}/>);
    expect(wrapper).toMatchSnapshot();
});

it('UndoList 为初始化列表, 数目为0， 列表无内容', () => {
    const listData = [];
    const wrapper = shallow(<UndoList list={listData}/>);
    const countEl = findTestWrapper(wrapper, 'count');
    const listItems = findTestWrapper(wrapper, 'list-item');
    expect(countEl.text()).toEqual('0');
    expect(listItems.length).toEqual(0);
});

it('UndoList 列表数据有内容，count显示长度，列表不为空', () => {
    const listData = ['vue', 'react', 'angular'];
    const wrapper = shallow(<UndoList list={listData}/>);
    const countEl = findTestWrapper(wrapper, 'count');
    const listItems = findTestWrapper(wrapper, 'list-item');
    expect(countEl.text()).toEqual('3');
    expect(listItems.length).toEqual(3)
});

it('UndoList 列表数据有内容，存在删除按钮', () => {
    const listData = ['vue', 'react', 'angular'];
    const wrapper = shallow(<UndoList list={listData}/>);
    const deleteItems = findTestWrapper(wrapper, 'delete-item');
    expect(deleteItems.length).toEqual(3);
});


it('UndoList 列表数据有内容，点击某个删除按钮，删除列表的某一项', () => {
    const listData = ['vue', 'react', 'angular'];
    const fn = jest.fn();
    const index = 1;
    const wrapper = shallow(<UndoList list={listData} deleteItem={fn}/>);
    const deleteItems = findTestWrapper(wrapper, 'delete-item');
    deleteItems.at(index).simulate('click');
    expect(fn).toHaveBeenLastCalledWith(index);
});

