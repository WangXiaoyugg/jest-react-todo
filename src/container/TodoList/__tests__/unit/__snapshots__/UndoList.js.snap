// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UndoList 渲染样式正常 1`] = `
<div>
  <div
    data-test="count"
  >
    0
  </div>
  <ul />
</div>
`;
